cpp/src/toppra/algorithm.cpp

```diff
@@ -5,6 +5,16 @@
 #include <iostream>
 #include "toppra/toppra.hpp"
 
+#include <ctime>
+#include <sys/time.h>
+
+unsigned long long GetCurrentTimeUsec()
+{
+  struct timeval tv;
+  gettimeofday(&tv,NULL);
+  return ((unsigned long long)tv.tv_sec * 1000000 + (unsigned long long)tv.tv_usec);
+}
```

Comment:
This is a C-style way of measuring time. Although valid, I would be in favor of a more C++ style (e.g. https://www.techiedelight.com/measure-elapsed-time-program-chrono-library).
I think `std::chrono` is also more portable.

Reply:
Ok, It's a good suggestion.